Freebsd docs

Zirias opened the pull request:
Hello Spiro,

Here are my suggestions for updating the docs, regarding issue #69 -- so far, there are changes to README.md and opencbm.sgml.

I open this PR now so anyone can already have a look, but will still think about how to maybe add the manpage for the FreeBSD kernel module :)

spiro-trikaliotis commented:
This looks good to me. It makes me aware that I have to add parts for Linux 4.x and Windows 10.
I'll wait for your further work.

The man page could be installed when installing the driver. Additionally, you could mention it in the opencbm/docs/opencbm.sgml file, so it is clear where to find info on the module.

Zirias commented:
Ok, added the man page now, but unfortunately, this is a hack that *might* break with newer FreeBSD versions. The FreeBSD build system doesn't allow installing man pages with kernel modules, so I added that by manually including `<bsd.man.mk>` and fiddling a bit with rules in dependencies. It will also fail at install time if the target directory for the man page doesn't exist yet -- but the same happens with the target directory of the module, so I guess this is ok. People should be adviced to use the official port anyways, which will take care of all these things :) But it probably makes sense to have the man page together with the code it describes.

Will add a hint about the manpage in the docs, and maybe remove the existing but outdated FreeBSD port, and then this PR should be complete -- feel free to comment and suggest improvements please :)

Zirias commented:
Well, would call it "done" from my side, but if you have comments, I can still change things :) I see some issues with the structure, cause the three bullet points about parallel cables, usb cables, and "both" ... don't really make sense. As I understand right now, the supported platforms of OpenCBM are Linux, FreeBSD, Windows and MacOS -- and now, only MacOS doesn't have support for parallel-port cables? So, maybe this should be refactored, but I'm not really sure...

spiro-trikaliotis commented on opencbm/sys/freebsd/.gitignore:
```diff
@@ -0,0 +1,9 @@
+!opencbm.4
+!Makefile
+!.gitignore
+
```
Is the .gitignore file really needed? Personally, I prefer to have an almost clean .gitignore, but have all the unnecessary files removed with "make mrproper".

Zirias replied:
Spiro, you can remove it, it is just a suggestion: I personally prefer to never have any files appear in `git status` that shouldn't be added to git. I know this is a matter of taste, so if you don't like it, get rid of it :)

Zirias replied:
If you prefer, I can rewrite the history of this PR without the .gitignore? I mean, for a clean commit? Or maybe you like my argumentation? ;)

spiro-trikaliotis replied:
It's ok for me.
The reason why I do not like the .gitignore: I do not see directly if the "mrproper" target works as expected.
But as it is "only" in the FreeBSD part, I could live with it, as I do not/cannot touch this.
